### PR TITLE
Add a distrib test for source-only ruby packages on linux

### PR DESCRIPTION
**Pull request by apolcyn**
This is a followup to the idea mentioned in https://github.com/grpc/grpc/issues/23593#issuecomment-664256161

Passing artifact/package/distrib build in https://g3c.corp.google.com/results/invocations/4b3a7f3d-da43-470f-ba9a-31a862ddf44e/targets

### CODE REVIEW

**Comment by apolcyn** [comment]
@jtattermusch distrib test is passing, this is ready for a look

File: test/distrib/ruby/run_distrib_test.sh [hunk 1]
```diff
@@ -17,12 +17,26 @@ set -ex
...
+  GEM_NAME="grpc-${GRPC_VERSION}.gem"
+else
+  [[ "$PACKAGE_TYPE" == "binary" ]] || die "unexpeced package type: $PACKAGE_TYPE"
+  GEM_NAME="grpc-${GRPC_VERSION}-${ARCH}-${PLATFORM}.gem"
+fi
 # Create an indexed local gem source with gRPC gems to test
 GEM_SOURCE=../../../gem_source
 mkdir -p "${GEM_SOURCE}/gems"
-cp "$EXTERNAL_GIT_ROOT"/input_artifacts/grpc-*"$ARCH-$PLATFORM".gem "${GEM_SOURCE}/gems"
+cp "${EXTERNAL_GIT_ROOT}/input_artifacts/${GEM_NAME}" "${GEM_SOURCE}/gems"
```
Comment (stanley-cheung) [inline test/distrib/ruby/run_distrib_test.sh:39]: So the source-only gem (the one above named `grpc-${GRPC_VERSION}.gem`) already exists (or, already built in a previous step)?

Comment (apolcyn) [reply test/distrib/ruby/run_distrib_test.sh:39]: Yep, it's built in the artifacts job that is chained to the distrib tests job as a pre-req, and passed to this job via this `EXTERNAL_GIT_ROOT/input_artifacts` directory.

**Comment by apolcyn** [comment]
C/C++ macos: https://github.com/grpc/grpc/issues/23669
interop cloud to prod: https://github.com/grpc/grpc/issues/23055

**Pull request merged**